base/applications/mspaint/dib.cpp

```diff
@@ -82,41 +82,111 @@ void ShowFileLoadError(LPCTSTR name)
     mainWindow.MessageBox(strText, strProgramName, MB_OK | MB_ICONEXCLAMATION);
 }
 
-void
-LoadDIBFromFile(HBITMAP * hBitmap, LPCTSTR name, LPSYSTEMTIME time, int *size, int *hRes, int *vRes)
+BOOL DoLoadImageFile(HWND hwnd, HBITMAP *phBitmap, LPCTSTR name, BOOL fIsMainFile)
 {
-    CImage img;
-    img.Load(name);
-    *hBitmap = img.Detach();
+    HBITMAP hBitmap = NULL;
 
-    if (!hBitmap)
+    if (phBitmap)
+        *phBitmap = NULL;
+
+    WIN32_FIND_DATAW find;
+    HANDLE hFind = FindFirstFileW(name, &find);
+    if (hFind == INVALID_HANDLE_VALUE)
     {
-        ShowFileLoadError(name);
-        return;
+        // does not exist
+        CStringW strText;
+        strText.Format(IDS_LOADERRORTEXT, name);
+        MessageBoxW(hwnd, strText, NULL, MB_ICONERROR);
+        return FALSE;
     }
+    FindClose(hFind);
 
-    // update time and size
-    HANDLE hFile =
-        CreateFile(name, GENERIC_READ, FILE_SHARE_READ, NULL, OPEN_EXISTING, FILE_FLAG_SEQUENTIAL_SCAN, NULL);
-    if (hFile == INVALID_HANDLE_VALUE)
+    // check the file size
+    DWORD dwFileSize = find.nFileSizeLow;
+    if (fIsMainFile)
     {
-        ShowFileLoadError(name);
-        return;
+        fileSize = dwFileSize;
+        fileHPPM = fileVPPM = 0;
     }
 
-    if (time)
+    // is file empty?
+    if (dwFileSize == 0)
+    {
+        if (fIsMainFile)
+        {
+            imageModel.ClearHistory();
+
+            GetFullPathName(name, SIZEOF(filepathname), filepathname, NULL);
+
+            CString strTitle;
+            strTitle.Format(IDS_WINDOWTITLE, PathFindFileName(filepathname));
+            mainWindow.SetWindowText(strTitle);
+
+            isAFile = TRUE;
+            registrySettings.SetMostRecentFile(filepathname);
+
+            return TRUE;
+        }
+    }
+
+    CImage img;
+    img.Load(name);
+    hBitmap = img.Detach();
+
+    if (hBitmap == NULL)
     {
+        // cannot open and not empty
```

Comment:
is comment still relevant for "empty" now that there is a dwFileSize == 0 check above ?

Reply:
OK.